### PR TITLE
MINOR: Rename `AlterIsrManager` to `AlterPartitionManager`

### PR DESCRIPTION
**Pull request by hachikuji**
Since we have changed the `AlterIsr` API to `AlterPartition`, it makes sense to rename `AlterIsrManager` as well.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by jsancio — APPROVED** [review]
LGTM.

File: core/src/test/scala/unit/kafka/cluster/AbstractPartitionTest.scala [hunk 1]
```diff
@@ -49,7 +49,7 @@ class AbstractPartitionTest {
   var logDir1: File = _
   var logDir2: File = _
   var logManager: LogManager = _
-  var alterIsrManager: MockAlterIsrManager = _
+  var alterIsrManager: MockAlterPartitionManager = _
   var isrChangeListener: MockIsrChangeListener = _
```
Comment (dengziming) [inline core/src/test/scala/unit/kafka/cluster/AbstractPartitionTest.scala]: Should we also change IsrChangeListener to PartitionChangeListener?

**Review by dengziming — COMMENTED** [review]
Thank you for this PR @hachikuji , left one question.

File: core/src/main/scala/kafka/server/AlterPartitionManager.scala [hunk 2]
```diff
@@ -61,12 +61,14 @@ trait AlterIsrManager {
...
 
-case class AlterIsrItem(topicPartition: TopicPartition,
-                        leaderAndIsr: LeaderAndIsr,
-                        future: CompletableFuture[LeaderAndIsr],
-                        controllerEpoch: Int) // controllerEpoch needed for Zk impl
+case class AlterPartitionItem(
+  topicPartition: TopicPartition,
+  leaderAndIsr: LeaderAndIsr,
+  future: CompletableFuture[LeaderAndIsr],
+  controllerEpoch: Int // controllerEpoch needed for Zk impl
```
Comment (dajac) [inline core/src/main/scala/kafka/server/AlterPartitionManager.scala]: nit: Should we say ZkAlterPartitionManager or ZK controller instead of « zk impl »?

**Review by dajac — COMMENTED** [review]
Renaming it makes sense. Left a small nit.

File: core/src/test/scala/unit/kafka/cluster/AbstractPartitionTest.scala (hunk 1 above)
Comment (hachikuji) [reply core/src/test/scala/unit/kafka/cluster/AbstractPartitionTest.scala]: Yeah, makes sense.

**Comment by hachikuji** [comment]
@dajac @dengziming Thanks for reviews. Pushed an update.

**Review by dajac — APPROVED** [review]
LGTM

File: core/src/test/scala/unit/kafka/server/KafkaServerTest.scala [hunk 3]
```diff
@@ -129,8 +129,8 @@ class KafkaServerTest extends QuorumTestHarness {
     props.put(KafkaConfig.InterBrokerProtocolVersionProp, ApiVersion.latestVersion.toString)
 
     val server = TestUtils.createServer(KafkaConfig.fromProps(props))
-    server.replicaManager.alterIsrManager match {
-      case _: DefaultAlterIsrManager =>
+    server.replicaManager.alterPartitionManager match {
```
Comment (dengziming) [inline core/src/test/scala/unit/kafka/server/KafkaServerTest.scala:132]: nit: testAlterIsrManager can also be renamed.

**Review by dengziming — APPROVED** [review]
LGTM, there are still 1 minor improvements: ApiKeysTest.testAlterIsrIsClusterAction.

**Review by dengziming — APPROVED** [review]

**Pull request merged**